Cargo's configure script doesn't mention that it requires a C compiler.

Later, `make` fails because it needs gcc to compile some of cargo's dependencies.
```
<snip>
target/snapshot/bin/cargo.exe build --target i686-pc-windows-gnu
   Compiling flate2 v0.0.1 (https://github.com/alexcrichton/flate2-rs#31d92533)
   Compiling toml v0.1.0 (https://github.com/alexcrichton/toml-rs#ac6f3b27)
   Compiling semver v0.1.0 (https://github.com/rust-lang/semver#7dca047a)
   Compiling libssh2-static-sys v0.0.1 (https://github.com/alexcrichton/libssh2-static-sys#c0dc19bd)
   Compiling encoding-index-tradchinese v1.0.20140915 (https://github.com/lifthrasiir/rust-encoding#84152708)
   Compiling encoding-index-simpchinese v1.0.20140915 (https://github.com/lifthrasiir/rust-encoding#84152708)
   Compiling encoding-index-japanese v1.0.20140915 (https://github.com/lifthrasiir/rust-encoding#84152708)
   Compiling docopt v0.6.8 (https://github.com/docopt/docopt.rs#0b184b15)
   Compiling tar v0.0.1 (https://github.com/alexcrichton/tar-rs#47d2cc4b)
   Compiling glob v0.0.1 (https://github.com/rust-lang/glob#a4695346)
   Compiling encoding-index-korean v1.0.20140915 (https://github.com/lifthrasiir/rust-encoding#84152708)
   Compiling openssl-static-sys v0.0.1 (https://github.com/alexcrichton/openssl-static-sys#2e1c50a7)
   Compiling encoding-index-singlebyte v1.0.20140915 (https://github.com/lifthrasiir/rust-encoding#84152708)
   Compiling link-config v0.0.1 (https://github.com/alexcrichton/link-config#7c56ef12)
   Compiling curl-sys v0.0.1 (https://github.com/alexcrichton/curl-rust?ref=bundle#b595431a)
   Compiling encoding v0.2.0 (https://github.com/lifthrasiir/rust-encoding#84152708)
   Compiling url v0.1.0 (https://github.com/servo/rust-url#cec3c321)
Build failed, waiting for other jobs to finish...
Failed to run custom build command for `flate2 v0.0.1 (https://github.com/alexcrichton/flate2-rs#31d92533)`
Process didn't exit successfully: `sh build/make.sh -C build` (status=2)
--- stdout
make[1]: Entering directory '/home/user/.cargo/git/checkouts/flate2-rs-2d49bad5a48eff46/master/build'
gcc -m32  -O2 miniz.c -c -o C:\NW\cargo\target\i686-pc-windows-gnu\native\flate2-8a239c8868e9f5b2/miniz.o
Makefile:14: recipe for target 'all' failed
make[1]: Leaving directory '/home/user/.cargo/git/checkouts/flate2-rs-2d49bad5a48eff46/master/build'

--- stderr
make[1]: gcc: Command not found
make[1]: *** [all] Error 127

Makefile:77: recipe for target 'cargo-i686-pc-windows-gnu' failed
make: *** [cargo-i686-pc-windows-gnu] Error 101
```

--- a/configure
+++ b/configure
@@ -342,6 +342,7 @@ step_msg "looking for build programs"
 
 probe_need CFG_CURLORWGET  curl wget
 probe_need CFG_PYTHON      python
+probe_need CFG_CC          cc gcc clang
 
 if [ ! -z "${CFG_LOCAL_RUST_ROOT}" ]; then
     CFG_RUSTDOC="${CFG_LOCAL_RUST_ROOT}/bin/rustdoc"

